### PR TITLE
Bug Fix for CloudEvent data access

### PR DESCRIPTION
**Pull request by mtickoobb**
**What this PR does / why we need it**:
Bug Fix for Cloud Event data access
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by mtickoobb** [comment]
@yuzisun

File: python/kfserving/test/test_server.py [hunk 1]
```diff
@@ -234,14 +244,32 @@ def app(self):  # pylint: disable=no-self-use
...
+        assert resp.headers['content-type'] == "application/x-www-form-urlencoded"
+        assert resp.headers['ce-specversion'] == "1.0"
+        assert resp.headers['ce-id'] == "36077800-0c23-4f38-a0b4-01f4369f670a"
+        assert resp.headers['ce-source'] == "https://example.com/event-producer"
+        assert resp.headers['ce-type'] == "com.example.sampletype1"
+        assert resp.headers['ce-datacontenttype'] == "application/x-www-form-urlencoded"
+        assert resp.headers['ce-time'] > "2021-01-28T21:04:43.144141+00:00"
 
+    async def test_predict_ce_binary_json(self, http_server_client):
+        event = dummy_cloud_event(json.loads('{"instances":[[1,2]]}'), set_contenttype=True)
```
Comment (yuzisun) [inline python/kfserving/test/test_server.py:266]: this test looks identical to the above one, `json.loads('{"instances":[[1,2]]}')` effectively is the same as `{"instances":[[1,2]]}`

File: python/kfserving/kfserving/kfmodel.py [hunk 2]
```diff
@@ -52,15 +53,27 @@ def load(self) -> bool:
...
-            return request
+        response = request
+
+        if(isinstance(request, CloudEvent)):
+            response = request.data
+            if(isinstance(response, bytes)):
+                try:
+                    response = json.loads(response.decode('UTF-8'))
+                except (json.JSONDecodeError):
+                    pass
```
Comment (yuzisun) [inline python/kfserving/kfserving/kfmodel.py:64]: I think we should still throw exception if the content type is set to `application/json` but not able to decode, if the content type is not set then we can pass.

File: python/kfserving/test/test_server.py (hunk 1 above)
Comment (mtickoobb) [reply python/kfserving/test/test_server.py:266]: Ya that's true. 
I think I'll just remove the test. 
I tried to add this to replicate the json CE issue.

**Comment by mtickoobb** [comment]
@yuzisun For some reason, I don't see your changes made in https://github.com/kubeflow/kfserving/commit/cac8d514b7c9e579cec7a7bb26c57a877215a339 on [README.md](https://github.com/kubeflow/kfserving/blob/master/README.md)

**Comment by yuzisun** [comment]
@mtickoobb Can you help do a rebase? Not sure why it includes my earlier commit.

**Comment by mtickoobb** [comment]
@yuzisun Please review now.

File: python/kfserving/kfserving/kfmodel.py (hunk 2 above)
Comment (mtickoobb) [reply python/kfserving/kfserving/kfmodel.py]: Done

**Comment by yuzisun** [comment]
Thanks @mtickoobb !

/lgtm
/approve

**Pull request merged**